pkg/rule/rule.go

```diff
@@ -316,12 +301,48 @@ func executeCommandAction(
 
 		// regardless of the outcome of running the background command, we first update the
 		// owner to track pods which might have a running background command
-		updateErr := updateRunningCommandPodListInOwner(owner, podsForAction, taskID.String())
+		podsForTracker := podsForAction
+
+		// Get pods already existing in tracker so we don't lose them
+		existingTracker, err := getPodsTrackerForOwner(owner)
+		if err != nil {
+			return err
+		}
+
+		if existingTracker != nil && len(existingTracker.Pods) > 0 {
+			for _, existingPod := range existingTracker.Pods {
+				// Check if pod exists in cluster
+				existingPodObject, err := k8s.Instance().GetPodByUID(types.UID(existingPod.UID), existingPod.Namespace)
+				if err != nil {
+					if err == k8s.ErrPodsNotFound {
+						continue
+					}
+
+					return err
+				}
+
+				// Check duplicates
+				skipPod := false
+				for _, pod := range podsForAction {
```

Comment:
Store the pods in a map with  UID->*podInfo so that you don't need this loop within a loop. After the loop put it in an array.

Reply:
done